osf/models/collection.py

```diff
@@ -39,6 +45,68 @@ class Meta:
     program_area = models.CharField(blank=True, max_length=127)
     school_type = models.CharField(blank=True, max_length=127)
     study_design = models.CharField(blank=True, max_length=127)
+    machine_state = models.IntegerField(
+        choices=CollectionSubmissionStates.int_field_choices(),
+        default=CollectionSubmissionStates.IN_PROGRESS,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.state_machine = CollectionSubmissionMachine(
+            model=self,
+            active_state=self.state,
+            state_property_name='state'
+        )
+
+    @property
+    def state(self):
+        '''Property to translate between ApprovalState Enum and DB string.'''
+        return CollectionSubmissionStates(self.machine_state)
+
+    @property
+    def is_moderated(self):
+        return self.provider and self.provider.reviews_workflow == 'post-moderation'
+
+    @state.setter
+    def state(self, new_state):
+        self.machine_state = new_state.value
+
+    def _on_submit(self, event_data):
+        self.submitted_timestamp = timezone.now()
+
+    def _on_accept(self, event_data):
+        user = event_data.kwargs['user']
+        if user.has_perm('accept_submissions', self.provider):
+            raise PermissionsError(f'{user} must have moderator permissions.')
+
+    def _on_reject(self, event_data):
+        user = event_data.kwargs['user']
+        if user.has_perm('reject_submissions', self.provider):
+            raise PermissionsError(f'{user} must have moderator permissions.')
+
+    def _on_remove(self, event_data):
+        pass
+
+    def _on_resubmit(self, event_data):
+        pass
+
+    def _save_transition(self, event_data):
+        '''Save changes here and write the action.'''
+        self.save()
+        from_state = CollectionSubmissionStates[event_data.transition.source]
+        to_state = self.state
+
+        trigger = CollectionSubmissionsTriggers.from_db_name(event_data.event.name)
+        if trigger is None:
+            return
```

Comment:
None of this is tested. Like, anywhere. Is that in a follow-up PR?

Reply:
This was just necessary to make the model changes, so the endpoint could be included, but there's no behavior to test that actually uses the state machine yet.

Reply:
I mean, the model logic exists, so you could test it in `osf_tests`